### PR TITLE
Enable Janus Admin Api as default on localhost

### PR DESCRIPTION
**Pull request by alepolidori**
NethServer/dev#6217
https://github.com/nethesis/dev/issues/5836

### CODE REVIEW

File: root/etc/httpd/conf.d/janus-admin-api.conf
Comment (DavidePrincipi) [inline root/etc/httpd/conf.d/janus-admin-api.conf]: What is this fragment? Why is it needed?

Isn't already possible to talk with the admin API on  http://127.0.0.1:7088?

Comment (gsanchietti) [reply root/etc/httpd/conf.d/janus-admin-api.conf]: I was about to ask the same thing :D

Comment (DavidePrincipi) [reply root/etc/httpd/conf.d/janus-admin-api.conf]: Maybe it's worth to add some examples of `curl` calls that interact with the administrative API...

Comment (alepolidori) [reply root/etc/httpd/conf.d/janus-admin-api.conf]: You are right ! :)
Once you have enabled the admin api, you can interact with them on `/admin` path.
Janus provides some demo pages on the above path.
Included into the demo pages, there is the admin.html page, that is an already developed ready to use page that provides some useful information (using admin apis), such as:
- janus version
- all the sessions with the associated plugin handles (useful to debug problems on ice,ssl,streams, etc...)
- you change the log level at runtime
- ...

You can take a look at the https://janus.conf.meetecho.com/admin.html
Without the admin.html page, you have to develop something similar.

Security aspects:
once you have open admin.html, you have to specify the admin url and the admin secret

Comment (DavidePrincipi) [reply root/etc/httpd/conf.d/janus-admin-api.conf]: I'm afraid it is quite complex to use that endpoint: it would require to port forward both 443 and 7088 TCP ports to use it?

Can we just document some `curl` calls? or write a little helper script for that?

Comment (alepolidori) [reply root/etc/httpd/conf.d/janus-admin-api.conf]: If nethvoice is reachable from outside, you have only to create a tunnel for 7088, otherwise you are right, you have to port forward also 443, but in this case I think it is necessary even if you don't need to debug janus.
I think also that the port forward is needed also with the usage of curl, with the difference that you have to create a script to check something: for example to have all the plugin sip handles of a specific session, you have to do 2 curl requests, instead of open the web page admin.html.

I think admin.html offers something ready to use to simplify some basic usage, but this is only my opinion: if you think that develop some additional scripts is a better way for me is ok.

Comment (DavidePrincipi) [reply root/etc/httpd/conf.d/janus-admin-api.conf]: - `curl` can be invoked on the local system: no port forwards are needed.
- If the ssh tunnel is just for port 7088, are we going to hit Same-Origin policy restrictions?

**Review by DavidePrincipi — APPROVED** [review]
I got the answer by myself in the rest of the config file: CORS are not enabled by default.

Ok to merge it. Please add a bit of documentation about how to connect to the admin interface: e.g.: set up the tunnel for 7088, connect URL http://localhost:7088, credentials (?)...

File: root/etc/httpd/conf.d/janus-admin-api.conf
Comment (alepolidori) [reply root/etc/httpd/conf.d/janus-admin-api.conf]: - No Same-Origin policy, example:
image

File: root/etc/httpd/conf.d/janus-admin-api.conf
Comment (gsanchietti) [inline root/etc/httpd/conf.d/janus-admin-api.conf]: This configuration opens a demo software to the public world, I don't think it's safe enough.
99% of the users will not use it and event will not be aware to have it publicly accessible.

Since this is a simple drop-in useful only for developers, I'd move it to the README.

**Review by gsanchietti — CHANGES_REQUESTED** [review]
I agree on opening the janus API from localhost, but I'm against the httpd configuration.

Also, bear in mind that, if you add httpd configuration, you need to also to:
- add httpd as dependencies
- restart httpd service inside createlinks

**Comment by alepolidori** [comment]
> I got the answer by myself in the rest of the config file: CORS are not enabled by default.
> 
> Ok to merge it. Please add a bit of documentation about how to connect to the admin interface: e.g.: set up the tunnel for 7088, connect URL http://localhost:7088, credentials (?)...

Here it is: https://github.com/NethServer/nethserver-janus/pull/29

I'am waiting for its approval, then I will merge this issue.
Is it ok @gsanchietti ?

**Comment by gsanchietti** [comment]
Maybe I have not been clear enough.
Totally remove the httpd configuration from the package, add to the README how to enable it.

**Comment by DavidePrincipi** [comment]
I'm not against distributing the UI part because as long as the API is not accessible it does not harm.

I also understand that it is useful only for developers: the README only solution proposed by Giacomo is good for me :+1:
The developer can also install the frontend part somewhere else (maybe also the Meetecho demo page is good?) and set the backend URI in the initial dialog...

File: root/etc/httpd/conf.d/janus-admin-api.conf
Comment (alepolidori) [reply root/etc/httpd/conf.d/janus-admin-api.conf]: Yes, you are right, I thought about that.
All the exposed demo require 8088 port to be open, but it is closed by default.
But I can agree that there is some code not useful for us that can expose some security aspects.
So, if you agree I can remove all the demos by removing httpd configuration and then developers can use admin api by curl, is it ok ?

Comment (gsanchietti) [reply root/etc/httpd/conf.d/janus-admin-api.conf]: Yes, it's ok for me! ;)

Comment (alepolidori) [reply root/etc/httpd/conf.d/janus-admin-api.conf]: Done :+1:

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**
